src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -217,6 +217,119 @@ .maxstack 8
             Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
         }
 
+        [Fact]
+        public void TestPropertyNotAssignedInStructConstructor()
+        {
+            var comp = CreateCompilation(@"
+var x = new C();
+x.P = 10;
+x = new C();
+
+public struct C
+{
+    public C()
+    {
+        System.Console.WriteLine(""In C..ctor: "" + P);
+    }
+
+    public int P { get => field; set => field = value; }
+}
+", options: TestOptions.ReleaseExe.WithSpecificDiagnosticOptions(ReportStructInitializationWarnings));
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            comp.VerifyDiagnostics(
+                // (10,51): warning CS9020: The 'this' object is read before all of its fields have been assigned, causing preceding implicit assignments of 'default' to non-explicitly assigned fields.
+                //         System.Console.WriteLine("In C..ctor: " + P);
+                Diagnostic(ErrorCode.WRN_UseDefViolationThisSupportedVersion, "P").WithArguments("this").WithLocation(10, 51)
+                );
+            CompileAndVerify(comp, expectedOutput: @"In C..ctor: 0
+In C..ctor: 0").VerifyIL("C..ctor", @"
+{
+    // Code size       37 (0x25)
+    .maxstack  2
+    .locals init (int V_0)
+    IL_0000:  ldarg.0
+    IL_0001:  ldc.i4.0
+    IL_0002:  stfld      ""int C.<P>k__BackingField""
+    IL_0007:  ldstr      ""In C..ctor: ""
+    IL_000c:  ldarg.0
+    IL_000d:  call       ""int C.P.get""
+    IL_0012:  stloc.0
+    IL_0013:  ldloca.s   V_0
+    IL_0015:  call       ""string int.ToString()""
+    IL_001a:  call       ""string string.Concat(string, string)""
+    IL_001f:  call       ""void System.Console.WriteLine(string)""
+    IL_0024:  ret
+}
+");
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetMembers().OfType<FieldSymbol>());
+            Assert.Equal(1, accessorBindingData.NumberOfPerformedAccessorBinding);
```

Comment:
>Assert.Equal(1, accessorBindingData.NumberOfPerformedAccessorBinding); [](http://example.com/codeflow?start=12&length=70)

Why do we need an extra binding for this scenario?

Reply:
#Pending

Reply:
It depends on which is compiled first, the constructor or the property.
If the constructor is compiled first, then definite assignment needs to the property backing field, which is unknown at this point so we force bind.
If property is compiled first then we calculate the backing field as part of the regular binding.

Meaning that the following doesn't trigger extra binding:

```csharp
public struct C
{
    public int P { get => field; set => field = value; }

    public C()
    {
        System.Console.WriteLine(""In C..ctor: "" + P);
    }
}
```

But the following does trigger extra binding

```csharp
public struct C
{
    public C()
    {
        System.Console.WriteLine(""In C..ctor: "" + P);
    }

    public int P { get => field; set => field = value; }
}
```

I can look into modifying `MethodCompiler.CompileNamedType` to force it bind accessors first. But I'm not sure if it's worth it. Let me know what do you think

Reply:
> But I'm not sure if it's worth it. Let me know what do you think

I think it is worth doing. I think some other checks that aren't implemented yet can benefit from this reordering too. We should avoid doing unnecessary work if there is a reasonable way to achieve that. Extra binding looks unnecessary in this scenario.